component/prometheus/exporter/consul/consul.go

```diff
@@ -21,9 +21,9 @@ func init() {
 	})
 }
 
-func createExporter(opts component.Options, args component.Arguments) (integrations.Integration, error) {
+func createExporter(opts component.Options, args component.Arguments, defaultInstanceKey string) (integrations.Integration, string, error) {
 	a := args.(Arguments)
-	return a.Convert().NewIntegration(opts.Logger)
+	return integrations.NewIntegrationWithInstanceKey(opts.Logger, a.Convert(), defaultInstanceKey)
```

Comment:
I'm confused by the API here. There is already a mechanism to customize target by using `NewWithTargetBuilder` as your Build function. How is this different form that? I don't love that a component that is fine with the default now has to accept and pass the default through. It feels like an unnecessary detail that should be hidden until they decide to override it.

Reply:
The use of the new `integrations.NewIntegrationWithInstanceKey` helper is to standardize calling the two `InstanceKey` and `NewIntegration` methods from the integrations' Config interface.

Since I'd like _all_ integrations to be able to call `InstanceKey()` to set their own instance label, I wanted to avoid using NewWithTargetBuilder everywhere, but I'm not against this approach if you think it's for the best.

Alternatively, how would you feel about removing the defaultInstance argument and always pass the result of `defaultInstance()` that resolves to the hostname?